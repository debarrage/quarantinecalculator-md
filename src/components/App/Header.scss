@import "./colors";

.app__header{
    &.header {    
        background-color: #ffffff11;
        color: $header-color;
        font-size: small;
        
        [class^='header__element'] {
            display: inline-block;

            height: 5vh;
            line-height: 5vh;
            padding-left: 1vh;

            a, span {
                margin-right: 1vw;
            }

            a {
                text-decoration: none;
                font-weight: bold;
                color: $header-color !important;
            }

            &[class*='--left'] {
                float: left;
            }

            &[class*='--right'] {
                float: right;
            }
        }
    }
}