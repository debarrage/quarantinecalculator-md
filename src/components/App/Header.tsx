import { MarkGithubIcon } from "@primer/octicons-react";
import React from "react";
import "./Header.scss";

/**
 * Header component
 */
export const Header: React.FC = () => {
    return (
        <div className="app__header header">
            <div className="header__element--left">
                <a href="https://github.com/debarrage/quarantinecalculator-md" target="_blank" rel="noopener noreferrer">
                    <MarkGithubIcon/> 
                </a>
                <span>
                    v{process.env.REACT_APP_VERSION}
                </span>
            </div>
            <div className="header__element--right">
                <a href={process.env.REACT_APP_PRIVACY_URL}>
                    Privacy
                </a>
            </div>
            <div className="header__element--right">
                <a href={process.env.REACT_APP_ISSUE_URL}>
                    Probleem melden
                </a>
            </div>
        </div>
    );
}
