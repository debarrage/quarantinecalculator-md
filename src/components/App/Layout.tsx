import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import { initDoneAction } from "../../store/actions";
import { Header } from "./Header";
import { useWindowSize } from "./hooks/useWindowSize";
import "./Layout.scss";

/**
 * App layout component. It uses the window size to set a fixed
 * height and width of the component.
 * 
 * @param props 
 */
export const Layout: React.FC = (props) => {
    
    const [width, height] = useWindowSize();
    const dispatch = useDispatch();


    useEffect(() => {
        dispatch(initDoneAction());
    }, [dispatch]);

    return (
        <div className="app" style={{ width: `${width}px`, height: `${height}px` }}>
            <Header/>
            <div className="app__body">
                {props.children}
            </div>
        </div>
    )
};
