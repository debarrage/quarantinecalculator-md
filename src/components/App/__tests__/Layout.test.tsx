import { createStore, Store } from "@reduxjs/toolkit";
import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { Header } from "../Header";
import { Layout } from "../Layout";

const ReduxProvider: React.FC<{store: Store}> = (props) => (
    <Provider store={props.store}>{props.children}</Provider>
)

describe("Base Layout classes", () => {
    it("Renders layout", () => {
        // Arrange
        const store = createStore(state => state);
    
        // Act
        const tree = renderer.create(<ReduxProvider store={store}><Layout/></ReduxProvider>).toJSON();
    
        // Assert
        expect(tree).toMatchSnapshot();
    });
    
    it("Renders footer", () => {
        // Arrange
        const store = createStore(state => state);
    
        // Act
        const tree = renderer.create(<ReduxProvider store={store}><Header/></ReduxProvider>).toJSON();
    
        // Assert
        expect(tree).toMatchSnapshot();
    });
});
