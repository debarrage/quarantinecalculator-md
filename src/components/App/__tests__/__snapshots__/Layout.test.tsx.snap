// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Layout classes Renders footer 1`] = `
<div
  className="app__header header"
>
  <div
    className="header__element--left"
  >
    <a
      href="https://github.com/debarrage/quarantinecalculator-md"
      rel="noopener noreferrer"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        className="octicon"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path fill-rule=\\"evenodd\\" d=\\"M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z\\"></path>",
          }
        }
        fill="currentColor"
        height={16}
        role="img"
        style={
          Object {
            "display": "inline-block",
            "userSelect": "none",
            "verticalAlign": "text-bottom",
          }
        }
        viewBox="0 0 16 16"
        width={16}
      />
    </a>
    <span>
      v
      localdev
    </span>
  </div>
  <div
    className="header__element--right"
  >
    <a
      href="https://debarrage.github.io/quarantinecalculator-md/privacy.html"
    >
      Privacy
    </a>
  </div>
  <div
    className="header__element--right"
  >
    <a
      href="https://github.com/debarrage/quarantinecalculator-md/issues"
    >
      Probleem melden
    </a>
  </div>
</div>
`;

exports[`Base Layout classes Renders layout 1`] = `
<div
  className="app"
  style={
    Object {
      "height": "768px",
      "width": "1024px",
    }
  }
>
  <div
    className="app__header header"
  >
    <div
      className="header__element--left"
    >
      <a
        href="https://github.com/debarrage/quarantinecalculator-md"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          className="octicon"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path fill-rule=\\"evenodd\\" d=\\"M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z\\"></path>",
            }
          }
          fill="currentColor"
          height={16}
          role="img"
          style={
            Object {
              "display": "inline-block",
              "userSelect": "none",
              "verticalAlign": "text-bottom",
            }
          }
          viewBox="0 0 16 16"
          width={16}
        />
      </a>
      <span>
        v
        localdev
      </span>
    </div>
    <div
      className="header__element--right"
    >
      <a
        href="https://debarrage.github.io/quarantinecalculator-md/privacy.html"
      >
        Privacy
      </a>
    </div>
    <div
      className="header__element--right"
    >
      <a
        href="https://github.com/debarrage/quarantinecalculator-md/issues"
      >
        Probleem melden
      </a>
    </div>
  </div>
  <div
    className="app__body"
  />
</div>
`;
